concurrency/DynamicFontDownloadHelper.java

```diff
@@ -0,0 +1,896 @@
+        } catch (Exception e) {
+            Log.error(TAG, "Exception in copyAndDeleteFontFiles", e);
+            return false;
+        } finally {
+            try {
+                if (inputStream != null) {
+                    inputStream.close();
+                }
+                if (outputStream != null) {
+                    outputStream.close();
+                }
+            } catch (Exception e) {
+                Log.error(TAG, "Error in closing streams", e);
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Helper method which will create a symbolic link between font fils in AES to internal font location.
+     *
+     * @param directory the directory of resource to scan
+     * @return true if create success, false otherwise.
+     */
+    private boolean createSymlinkAtInternalFileSystem(@Nonnull File directory) {
+        if (!directory.exists() || !directory.isDirectory()) {
+            Log.error(TAG, " The file " + directory.getName() + " is not a directory or doesnt exist");
+            return false;
+        }
+
+        File[] files = directory.listFiles();
+
+        if (files != null) {
+            for (File f : files) {
+                if (!createSymlinkForFontFile(f)) {
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Helper method which will copy all the files in the input directory to internal font location
+     * @param directory, the directory to scan
+     * @param, resourceName, the name of the resource for which we are moving the files
+     * @return, true if copy succeeded, false otherwise
+     */
+    private boolean moveFontsToInternalFileSystem(@Nonnull File directory, @Nonnull String resourceName, @Nonnull String language) {
+        boolean hasError = false;
+        if (!directory.exists() || !directory.isDirectory()) {
+            Log.error(TAG, " The file " + directory.getName() + " is not a directory or doesnt exist");
+            return false;
+        }
+
+        File[] files = directory.listFiles();
+
+        boolean copyFonts = false;
+
+        if (files != null) {
+            for (File f : files) {
+                //Currently we can just return if font file already exist.When we get a newer version of fonts
+                //then we need to remove this If and support replacing the existing file. However font upgrade is a
+                //rare scenario.
+                if (doesFontFileExist(f.getName(), language)) {
+                    //File already exist lets delete the app expan file
+                    f.delete();
+                    continue;
+                }
+
+                //Even if one file doesnt exist already then we need to copy from downloaded path
+                copyFonts = true;
+
+                if (copyFontToInternalFileSystem(f)) {
+                    //Lets delete the font file from app expan directory
+                    f.delete();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.